src/test/kotlin/com/atlassian/performance/tools/virtualusers/lib/sshubuntu/SudoSshUbuntuImage.kt

```diff
@@ -0,0 +1,89 @@
+package com.atlassian.performance.tools.virtualusers.lib.sshubuntu
+
+import com.atlassian.performance.tools.ssh.api.Ssh
+import com.atlassian.performance.tools.ssh.api.SshHost
+import com.atlassian.performance.tools.ssh.api.auth.PasswordAuthentication
+import com.atlassian.performance.tools.virtualusers.lib.docker.ConnectedContainer
+import com.atlassian.performance.tools.virtualusers.lib.docker.CreatedContainer
+import com.atlassian.performance.tools.virtualusers.lib.docker.StartedDockerContainer
+import com.atlassian.performance.tools.virtualusers.lib.docker.execAsResource
+import com.github.dockerjava.api.DockerClient
+import com.github.dockerjava.api.model.ExposedPort
+import com.github.dockerjava.api.model.HostConfig
+import com.github.dockerjava.core.command.PullImageResultCallback
+import java.time.Duration
+
+class SudoSshUbuntuImage(
+    private val docker: DockerClient,
+    private val networkId: String,
+    private val portsToExpose: List<Int>
+) {
+
+    fun <T> runInUbuntu(
+        lambda: (SudoSshUbuntuContainer) -> T
```

Comment:
Does it have to be (SudoSshUbuntuContainer) -> T? Wouldn't be (SshConnection) -> T be enough?

Reply:
`SshConnection` is not enough, because `LoadTestIT` needs to:
* dig out the [`publishedJiraPort`](https://github.com/dagguh/virtual-users/blob/3416f2ea749bac9aa22148a4d7e7a4251dcc1035/src/test/kotlin/com/atlassian/performance/tools/virtualusers/LoadTestIT.kt#L120)
* pass in the [`db.peerIp`](https://github.com/dagguh/virtual-users/blob/3416f2ea749bac9aa22148a4d7e7a4251dcc1035/src/test/kotlin/com/atlassian/performance/tools/virtualusers/LoadTestIT.kt#L126)

Reply:
Different networks have different "address books". Ie. a host talking to containers uses different address space than containers talking to each other.